### PR TITLE
Adding layouts for ML

### PR DESCRIPTION
**Pull request by mfrachet**
- Add base layout for ML
- Add empty states for ML
- Add tests for empty states for ML

### CODE REVIEW

File: packages/core/helper-plugin/lib/src/components/AnErrorOccurred/AnErrorOccurred.stories.mdx [hunk 1]
```diff
@@ -0,0 +1,31 @@
+<!--- EmptyStateLayout.stories.mdx --->
```
Comment (soupette) [inline packages/core/helper-plugin/lib/src/components/AnErrorOccurred/AnErrorOccurred.stories.mdx:1]: `<!--- AnErrorOccured.stories.mdx --->`

File: packages/core/helper-plugin/lib/src/components/AnErrorOccurred/index.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        content.values
+      )}
+      {...rest}
+    />
+  );
+};
+
+AnErrorOccurred.defaultProps = {
+  content: {
+    id: 'anErrorOccurred',
```
Comment (soupette) [inline packages/core/helper-plugin/lib/src/components/AnErrorOccurred/index.js:24]: 👍

File: packages/core/upload/admin/src/pages/App/index.js [hunk 3]
```diff
@@ -1,31 +1,122 @@
...
+  NoMedia,
+  AnErrorOccurred,
+} from '@strapi/helper-plugin';
+import { Layout, HeaderLayout, ContentLayout, ActionLayout } from '@strapi/parts/Layout';
+import { Main } from '@strapi/parts/Main';
+import { Button } from '@strapi/parts/Button';
+import AddIcon from '@strapi/icons/AddIcon';
+import { Box } from '@strapi/parts/Box';
+import { BaseCheckbox } from '@strapi/parts/BaseCheckbox';
+import { Search } from './components/Search';
```
Comment (soupette) [inline packages/core/upload/admin/src/pages/App/index.js:19]: You can use the Search component from the helper plugin instead, it is not wrapped in the SearchForm one but you can make an update if needed.

File: packages/core/upload/admin/src/pages/App/components/Search.js [hunk 4]
```diff
@@ -0,0 +1,76 @@
+import React, { useState } from 'react';
```
Comment (soupette) [inline packages/core/upload/admin/src/pages/App/components/Search.js:1]: You can use the Search component from the helper plugin instead see the comment below.

File: packages/core/upload/admin/src/pages/App/index.js [hunk 5]
```diff
@@ -1,31 +1,122 @@
...
+    <Layout>
+      <Main aria-busy={loading}>
+        <HeaderLayout
+          title={formatMessage({
+            id: getTrad('plugin.name'),
+            defaultMessage: 'Media Library',
+          })}
+          subtitle={formatMessage(
+            {
+              id: getTrad('header.content.assets-multiple'),
```
Comment (soupette) [inline packages/core/upload/admin/src/pages/App/index.js:57]: Here you can create a new translation that handles the singular and the plural forms directly or use both the `header.content.assets.assets-multiple` or `header.content.assets.assets-single` ones.

We will clean the translations when the migration is over.

In case you decide to create a new one:

```
"header.content.label" : "{number, plural, =0 {# assets} one {# asset} other {# assets}}"
```

File: packages/core/upload/admin/src/hooks/useAssets.js [hunk 6]
```diff
@@ -0,0 +1,50 @@
+import { useEffect } from 'react';
+import { useQuery } from 'react-query';
+import { useNotifyAT } from '@strapi/parts/LiveRegions';
+import { useQuery as useURLQuery, useNotification } from '@strapi/helper-plugin';
```
Comment (soupette) [inline packages/core/upload/admin/src/hooks/useAssets.js:4]: Here you can use the `useQueryParams` instead, it will be easier later to handle the filters and the search.

File: packages/core/upload/admin/src/hooks/useAssets.js [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+import { useIntl } from 'react-intl';
+import { axiosInstance, getRequestUrl, generateStringFromParams } from '../utils';
+import useSelectTimestamps from './useSelectTimestamps';
+
+export const useAssets = ({ skipWhen }) => {
+  const { formatMessage } = useIntl();
+  const toggleNotification = useNotification();
+  const [, updated_at] = useSelectTimestamps();
+  const { notifyStatus } = useNotifyAT();
+  const query = useURLQuery();
```
Comment (soupette) [inline packages/core/upload/admin/src/hooks/useAssets.js:14]:
```
const [{ query, rawQuery }] = useQueryParams();

console.log(query) =>  // { sort: "updated_at:DESC" }
console.log(rawQuery) => // ?sort=updated_at:DESC

File: packages/core/upload/admin/src/hooks/useAssets.js [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+  const dataRequestURL = getRequestUrl('files');
+  const params = generateStringFromParams(query);
```
Comment (soupette) [inline packages/core/upload/admin/src/hooks/useAssets.js:16]: You can delete this utility as we will use `qs` and the values returned by the useQueryParams hook.

**Review by soupette — CHANGES_REQUESTED** [review]

File: packages/core/upload/admin/src/pages/App/index.js (hunk 3 above)
Comment (soupette) [reply packages/core/upload/admin/src/pages/App/index.js:19]: Also what since the timestamps are not configurable with the v4, what do you think of settings the url's parameters directly so you won't have to handle any logic in your child components:

``` 
const [{ query }, setQuery] = useQueryParams();

  React.useEffect(() => {
    if (!query) {
      setQuery({ sort: 'updatedAt:DESC', page: 1, pageSize: 10 });
    }
  }, [query, setQuery]);
```

File: packages/core/upload/admin/src/hooks/useAssets.js [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+  const paramsToSend = params.includes('sort') ? params : params.concat(`&sort=${updated_at}:DESC`);
```
Comment (soupette) [inline packages/core/upload/admin/src/hooks/useAssets.js:17]: This won't be needed anymore.

File: packages/core/upload/admin/src/hooks/useAssets.js [hunk 10]
```diff
@@ -0,0 +1,50 @@
...
+
+  const { data, error, isLoading } = useQuery(
+    'assets',
```
Comment (soupette) [inline packages/core/upload/admin/src/hooks/useAssets.js:20]:
```
['assets', rawQuery]
```

File: packages/core/upload/admin/src/hooks/useAssets.js [hunk 11]
```diff
@@ -0,0 +1,50 @@
...
+    async () => {
+      const { data } = await axiosInstance.get(`${dataRequestURL}?${paramsToSend}`);
```
Comment (soupette) [inline packages/core/upload/admin/src/hooks/useAssets.js:22]: `const { data } = await axiosInstance.get(`${dataRequestURL}${rawQuery}`);`

**Review by soupette — CHANGES_REQUESTED** [review]

File: packages/core/upload/admin/src/hooks/useAssets.js (hunk 10 above)
Comment (mfrachet) [reply packages/core/upload/admin/src/hooks/useAssets.js:20]: why?

Comment (soupette) [reply packages/core/upload/admin/src/hooks/useAssets.js:16]: According to the docs it's easier to make the pagination, you can prefetch the next page and the page before this way.

Comment (mfrachet) [reply packages/core/upload/admin/src/hooks/useAssets.js:16]: Let's to this in the pagination PR if it's okay for you

**Review by soupette — APPROVED** [review]
LGTM
